### PR TITLE
[plugin.video.gamekings] 1.2.15

### PR DESCRIPTION
**Pull request by skipmodea1**
### Description
v1.2.15 (2019-03-01)
- vimeo disabled oauth1, causing the vimeo addon to break. Therefore not using that addon anymore
- improved detecting if the video is a twitch live-stream
- changed fanart
- switch to Leia (kodi 18)
- removed gamekings extra category
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: plugin.video.gamekings/addon.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+if len(sys.argv[2]) == 0:
+    #
+    # Main menu
+    #
+    xbmc.log("[ADDON] %s, Python Version %s" % (ADDON, str(sys.version)), xbmc.LOGDEBUG)
+    xbmc.log("[ADDON] %s v%s (%s) is starting, ARGV = %s" % (ADDON, VERSION, DATE, repr(sys.argv)),
+                 xbmc.LOGDEBUG)
+
+
+    if SETTINGS.getSetting('onlyshowvideoscategory') == 'true':
```
Comment (Rechi) [inline plugin.video.gamekings/addon.py]:
```suggestion
    if SETTINGS.getSettingBool('onlyshowvideoscategory'):
```

File: plugin.video.gamekings/resources/lib/gamekings_list.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    # Init
+    #
+    def __init__(self):
+        # Get the command line arguments
+        # Get the plugin url in plugin:// notation
+        self.plugin_url = sys.argv[0]
+        # Get the plugin handle as an integer number
+        self.plugin_handle = int(sys.argv[1])
+
+        log("ARGV", repr(sys.argv))
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_list.py]: Kodi logs that already

File: plugin.video.gamekings/resources/lib/gamekings_list.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+            item = convertToUnicodeString(item)
+
+            # if item contains 'postcontainer, skip the item
+            if str(item).find('postcontainer') >= 0:
+
+                log("skipped item containing 'postcontainer'", item)
+
+                continue
+
+            log("item", item)
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_list.py]: log spam

File: plugin.video.gamekings/resources/lib/gamekings_list.py [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+            video_page_url = item.a['href']
+
+            log("video_page_url", video_page_url)
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_list.py]: log spam

File: plugin.video.gamekings/resources/lib/gamekings_list.py [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+            title = title.replace("Gamekings Extra: ", "")
+            title = title.replace("Gamekings Extra over ", "")
+            title = title.replace("Extra: ", "")
+            title = title.replace("Extra over ", "")
+            title = title.capitalize()
+
+            if str(item).find("title--premium") >= 0:
+                title = PREMIUM_ONLY_VIDEO_TITLE_PREFIX + ' ' + title
+
+            log("title", title)
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_list.py]: log spam

File: plugin.video.gamekings/resources/lib/gamekings_list.py [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+            try:
+                thumbnail_url = item.a.img['data-original']
+            except:
+                # skip the item if it has no thumbnail
+
+                log("skipping item with no thumbnail", item)
+
+                continue
+
+            log("thumbnail_url", thumbnail_url)
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_list.py]: log spam

File: plugin.video.gamekings/resources/lib/gamekings_play.py [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+        # Get the command line arguments
+        # Get the plugin url in plugin:// notation
+        self.plugin_url = sys.argv[0]
+        # Get the plugin handle as an integer number
+        self.plugin_handle = int(sys.argv[1])
+
+        # Get plugin settings
+        self.IS_PREMIUM_MEMBER = SETTINGS.getSetting('is-premium-member')
+
+        log("ARGV", repr(sys.argv))
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_play.py]: Kodi logs that already

File: plugin.video.gamekings/resources/lib/gamekings_play.py [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+
+        #log("html_source[start_pos_video_url:]", html_source[start_pos_video_url:])
+
+        # Try to make a valid video url
+        if have_valid_url:
+            # Let's only use the video_url part
+            html_source_split = str(html_source[start_pos_video_url:]).split()
+            video_url = html_source_split[0]
+
+            log("video_url after split", video_url)
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_play.py]: log spam

File: plugin.video.gamekings/resources/lib/gamekings_play.py [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+            elif video_url.find("player.vimeo.com/api/player.js") >= 0:
+                no_url_found = True
+                have_valid_url = False
+                video_url = ""
+            elif video_url.find("www.youtube.com/user/Gamekingsextra") >= 0:
+                no_url_found = True
+                have_valid_url = False
+                video_url = ""
+
+        log("video_url", video_url)
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_play.py]: log spam

File: plugin.video.gamekings/resources/lib/gamekings_search.py [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+    # Init
+    #
+    def __init__(self):
+        # Get the command line arguments
+        # Get the plugin url in plugin:// notation
+        self.plugin_url = sys.argv[0]
+        # Get the plugin handle as an integer number
+        self.plugin_handle = int(sys.argv[1])
+
+        log("ARGV", repr(sys.argv))
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_search.py]: Kodi logs that already

File: plugin.video.gamekings/resources/lib/gamekings_search.py [hunk 11]
```diff
@@ -0,0 +1,269 @@
...
+            item = convertToUnicodeString(item)
+
+            # if item contains 'postcontainer, skip the item
+            if str(item).find('postcontainer') >= 0:
+
+                log("skipped item containing 'postcontainer'", item)
+
+                continue
+
+            log("item", item)
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_search.py]: log spam

File: plugin.video.gamekings/resources/lib/gamekings_search.py [hunk 12]
```diff
@@ -0,0 +1,269 @@
...
+
+            video_page_url = item.a['href']
+
+            log("video_page_url", video_page_url)
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_search.py]: log spam

File: plugin.video.gamekings/resources/lib/gamekings_search.py [hunk 13]
```diff
@@ -0,0 +1,269 @@
...
+            title = title.replace("Gamekings Extra: ", "")
+            title = title.replace("Gamekings Extra over ", "")
+            title = title.replace("Extra: ", "")
+            title = title.replace("Extra over ", "")
+            title = title.capitalize()
+
+            if str(item).find("title--premium") >= 0:
+                title = PREMIUM_ONLY_VIDEO_TITLE_PREFIX + ' ' + title
+
+            log("title", title)
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_search.py]: log spam

File: plugin.video.gamekings/resources/lib/gamekings_search.py [hunk 14]
```diff
@@ -0,0 +1,269 @@
...
+            try:
+                thumbnail_url = item.a.img['data-original']
+            except:
+                # skip the item if it has no thumbnail
+
+                log("skipping item with no thumbnail", item)
+
+                continue
+
+            log("thumbnail_url", thumbnail_url)
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_search.py]: log spam

File: plugin.video.gamekings/resources/lib/gamekings_play.py [hunk 15]
```diff
@@ -0,0 +1,280 @@
...
+    #
+    def __init__(self):
+        # Get the command line arguments
+        # Get the plugin url in plugin:// notation
+        self.plugin_url = sys.argv[0]
+        # Get the plugin handle as an integer number
+        self.plugin_handle = int(sys.argv[1])
+
+        # Get plugin settings
+        self.IS_PREMIUM_MEMBER = SETTINGS.getSetting('is-premium-member')
```
Comment (Rechi) [inline plugin.video.gamekings/resources/lib/gamekings_play.py]:
```suggestion
        self.IS_PREMIUM_MEMBER = SETTINGS.self.Addon.getSettingBool('is-premium-member')
```

File: plugin.video.gamekings/addon.py (hunk 1 above)
Comment (skipmodea1) [reply plugin.video.gamekings/addon.py]: Fixed. I'll make a separate pr to set the addon to broken.

File: plugin.video.gamekings/resources/lib/gamekings_list.py (hunk 2 above)
Comment (skipmodea1) [reply plugin.video.gamekings/resources/lib/gamekings_list.py]: Fixed.

File: plugin.video.gamekings/resources/lib/gamekings_list.py (hunk 3 above)
Comment (skipmodea1) [reply plugin.video.gamekings/resources/lib/gamekings_list.py]: Fixed.

File: plugin.video.gamekings/resources/lib/gamekings_list.py (hunk 5 above)
Comment (skipmodea1) [reply plugin.video.gamekings/resources/lib/gamekings_list.py]: Fixed.

File: plugin.video.gamekings/resources/lib/gamekings_search.py (hunk 12 above)
Comment (skipmodea1) [reply plugin.video.gamekings/resources/lib/gamekings_search.py]: Fixed.

File: plugin.video.gamekings/resources/lib/gamekings_search.py (hunk 13 above)
Comment (skipmodea1) [reply plugin.video.gamekings/resources/lib/gamekings_search.py]: Fixed.

File: plugin.video.gamekings/resources/lib/gamekings_search.py (hunk 14 above)
Comment (skipmodea1) [reply plugin.video.gamekings/resources/lib/gamekings_search.py]: Fixed.

File: plugin.video.gamekings/resources/lib/gamekings_play.py (hunk 15 above)
Comment (skipmodea1) [reply plugin.video.gamekings/resources/lib/gamekings_play.py]: Fixed.

File: plugin.video.gamekings/resources/lib/gamekings_search.py (hunk 11 above)
Comment (skipmodea1) [reply plugin.video.gamekings/resources/lib/gamekings_search.py]: Fixed.

File: plugin.video.gamekings/resources/lib/gamekings_list.py (hunk 4 above)
Comment (skipmodea1) [reply plugin.video.gamekings/resources/lib/gamekings_list.py]: Fixed.

**Comment by Rechi** [comment]
@skipmodea1 I'll wait for the broken PR before merging this one. You may want to increment the version in this PR, so you can use `1.2.14` for the broken one exclusively.

**Comment by skipmodea1** [comment]
Not a clue on how to make that broken 1.2.14 pr.

>git checkout -b b-plugin.video.gamekings-1.2.14 plugin.video.gamekings/master
>fatal: A branch named 'b-plugin.video.gamekings-1.2.14' already exists.

**Pull request merged**